### DISCUSSION TITLE
Request for Improvement of Emulatorjs Functionality on Xbox One (Controller Problem)

### DISCUSSION THREAD

**Discussion by sagaksagak** [EmulatorJS/EmulatorJS#902] [General]
After installing the Docker version of emulatorjs on my NAS, I enjoyed games through the Xbox One's Edge browser. The Edge browser has a feature to switch the Xbox controller between web browsing mode and game controller mode. This can be toggled by pressing the ▤ button (three-line button, normally the start button). Since the library selection screen in emulatorjs is a web browsing screen before launching a game, if you press that screen and switch modes, the game runs in full screen, and all players (1P, 2P) can use the Xbox One controller normally. Alternatively, you can switch modes at any time during the game by pressing the start button.

However, with the recent GitHub version 4.1.1, if you switch to controller mode in an empty space of the Edge browser after pressing the play button in the game library, it doesn't work in the game. Or if it does work, only 1P functions while 2P doesn't. If you press the game controller settings button located in the bottom left of the game screen and enter, 1P has button mapping, but 2P doesn't. All slots are empty for 2P.
Additionally, when you press the start button on the game screen, a small dialog (screenshot/recording/save/load dialog) pops up, and the controller switch menu doesn't appear. This makes it very inconvenient. I discovered this situation in RomM and inquired about it in RomM's Discord channel, but I received the opinion that emulatorjs needs to make improvements.

I would greatly appreciate it if you could pay a bit more attention to controller support on Xbox One.

**Comment by ethanaobrien** [comment]
>After installing the Docker version of emulatorjs on my NAS

There is no docker version of this project, can you please give [this readme](https://github.com/linuxserver/docker-emulatorjs?tab=readme-ov-file#application-setup) a read and confirm you are in the right place? You can see if this is the correct project by going to https://demo.emulatorjs.org

>However, with the recent GitHub version 4.1.1, if you switch to controller mode in an empty space of the Edge browser after pressing the play button in the game library, it doesn't work in the game.

This isn't possible, no changes to HTML layouts were changed in 4.1.1: https://github.com/EmulatorJS/EmulatorJS/compare/v4.0.12...v4.1.1

>1P has button mapping, but 2P doesn't. All slots are empty for 2P.

Thats intended. It's intended to be set manually for Players 2-4

>Additionally, when you press the start button on the game screen, a small dialog (screenshot/recording/save/load dialog) pops up, and the controller switch menu doesn't appear.

This sounds like the browser isnt even being told this button is being pressed, its likely an xbox "feature" that cannot be changed



I do not personally own an Xbox one, any feature improvements would be up to you to implement

**Reply by sagaksagak** [reply]
I thought this is the docker version. https://hub.docker.com/r/linuxserver/emulatorjs
That version operations fine on xbox one with full controller support.
  

> > After installing the Docker version of emulatorjs on my NAS
> 
> There is no docker version of this project, can you please give [this readme](https://github.com/linuxserver/docker-emulatorjs?tab=readme-ov-file#application-setup) a read and confirm you are in the right place? You can see if this is the correct project by going to https://demo.emulatorjs.org
> 
> > However, with the recent GitHub version 4.1.1, if you switch to controller mode in an empty space of the Edge browser after pressing the play button in the game library, it doesn't work in the game.
> 
> This isn't possible, no changes to HTML layouts were changed in 4.1.1: [v4.0.12...v4.1.1](https://github.com/EmulatorJS/EmulatorJS/compare/v4.0.12...v4.1.1)
> 
> > 1P has button mapping, but 2P doesn't. All slots are empty for 2P.
> 
> Thats intended. It's intended to be set manually for Players 2-4
> 
> > Additionally, when you press the start button on the game screen, a small dialog (screenshot/recording/save/load dialog) pops up, and the controller switch menu doesn't appear.
> 
> This sounds like the browser isnt even being told this button is being pressed, its likely an xbox "feature" that cannot be changed
> 
> I do not personally own an Xbox one, any feature improvements would be up to you to implement

**Reply by sagaksagak** [reply]
If you remove the feature that creates a small window when pressing the start button on the game screen, it seems like it would work without issues on Xbox One.
Additionally, it would be great if the basic controller mappings were set up for P1 through P4 by default.

> > After installing the Docker version of emulatorjs on my NAS
> 
> There is no docker version of this project, can you please give [this readme](https://github.com/linuxserver/docker-emulatorjs?tab=readme-ov-file#application-setup) a read and confirm you are in the right place? You can see if this is the correct project by going to https://demo.emulatorjs.org
> 
> > However, with the recent GitHub version 4.1.1, if you switch to controller mode in an empty space of the Edge browser after pressing the play button in the game library, it doesn't work in the game.
> 
> This isn't possible, no changes to HTML layouts were changed in 4.1.1: [v4.0.12...v4.1.1](https://github.com/EmulatorJS/EmulatorJS/compare/v4.0.12...v4.1.1)
> 
> > 1P has button mapping, but 2P doesn't. All slots are empty for 2P.
> 
> Thats intended. It's intended to be set manually for Players 2-4
> 
> > Additionally, when you press the start button on the game screen, a small dialog (screenshot/recording/save/load dialog) pops up, and the controller switch menu doesn't appear.
> 
> This sounds like the browser isnt even being told this button is being pressed, its likely an xbox "feature" that cannot be changed
> 
> I do not personally own an Xbox one, any feature improvements would be up to you to implement

**Reply by ethanaobrien** [reply]
>If you remove the feature that creates a small window when pressing the start button on the game screen, it seems like it would work without issues on Xbox One.

This "feature" is only enabled when the owner of the website tells it to be enabled

> Additionally, it would be great if the basic controller mappings were set up for P1 through P4 by default.

This isn't planned. There are hundreds upon hundreds of different controllers out there, and I can't possibly assume a default controller mapping for every controller out there. You can simply either host if yourself and change the default mapping or set them in the menu when you first load the game

**Reply by sagaksagak** [reply]
1. oh great!!!  I am running my own synology nas. please let me know how to disable popup window so that le the edge browser know " long press start button."
2. avtually, I am trying. github:sagaksagak/romm_for_xbox but...not good yet. I am a newbie...on the programming field...
